compiler/luci/pass/src/RemoveUnnecessaryReshapePass.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+bool remove_no_effect_reshape(luci::CircleNode *node)
+{
+  auto target_node = dynamic_cast<luci::CircleReshape *>(node);
+  if (target_node == nullptr)
+    return false;
+
+  std::vector<int32_t> shape_info;
+  auto new_shape = dynamic_cast<luci::CircleConst *>(target_node->shape());
+  // If shape node is not a const node, get updated shape info from option.
+  if (new_shape == nullptr)
+  {
+    auto dummy_shape = dynamic_cast<luci::CircleOutputDummy *>(target_node->shape());
+    if (dummy_shape == nullptr)
+      return false;
+
+    // Need to extract new_shape info on target_node->newShape()
```

Comment:
@struss 
AFAIK, `CircleOutputDummy` is inserted when `Reshape` op doesn't have its option, which means it doesn't have `newShape` attribute as well. This line can be run?

Reply:
@mhs4670go sorry for my misunderstanding. I think your right. from `luci/import` logic, if second input not comes, shape is created from option. so if `CircleOutputDummy` comes in shape, it have neither shape and reshape option. 
I will fix this matching with `luci/import` logic. Thanks for pointing out!

Reply:
and also, if shape is `CircleOutputDummy`, then options also no value, follow from `luci/import` (can't extract shape from option)
https://github.com/Samsung/ONE/blob/5b4a868d686fe4d19af0c99921f784de8e027504/compiler/luci/import/src/Nodes/CircleReshape.cpp#L68-L76